modules/azure/client_factory.go

```diff
@@ -919,6 +916,35 @@ func CreateDataFactoriesClientE(subscriptionID string) (*datafactory.FactoriesCl
 	return &dataFactoryClient, nil
 }
 
+// CreatePrivateDnsZonesClientE is a helper function that will setup a private DNS zone client.
+func CreatePrivateDnsZonesClientE(subscriptionID string) (*privatedns.PrivateZonesClient, error) {
+	// Validate Azure subscription ID
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
```

Comment:
I would suggest changing subscriptionID to different name to avoid shadowing

Reply:
Renamed